hack/generate.sh

```diff
@@ -17,8 +17,39 @@
 set -e
 
 WHAT="protobuf codegen manifests logcheck gomegacheck monitoring-docs"
-WHICH="charts cmd example extensions pkg plugin test"
+CODEGEN_GROUPS=""
+MANIFESTS_FOLDERS=""
 MODE="parallel"
+AVAILABLE_CODEGEN_OPTIONS=(
+  "authentication"
+  "core"
+  "extensions"
+  "resources"
+  "operator"
+  "seedmanagement"
+  "operations"
+  "settings"
+  "operatorconfig"
+  "controllermanager"
+  "admissioncontroller"
+  "scheduler"
+  "gardenlet"
+  "resourcemanager"
+  "shoottolerationrestriction"
+  "shootdnsrewriting"
+  "provider_local"
+  "extensions_config"
```

Comment:
I would prefer 

```bash
AVAILABLE_CODEGEN_OPTIONS=(
  "authentication_groups"
  "core_groups"
  ...
)
```

to keep the things simple and obvious. Otherwise it is a hidden thingy that `_group` is appended in the update-codegen.sh script and the reader wonders how this options here correlate to the funcs in the update-codegen.sh script. At the end, we are not saving a lot of typing/duplication.